### PR TITLE
Move to SBT Pack instead of Assembly

### PR DESCRIPTION
**Pull request by idreeskhan**
#78 
#77 
#76 
and potentially #67 

We want to be able to move to SBT Pack instead of assembly to simplify the dependency management and avoid conflicts with building the fat jar. To make the usage easier, we reduce to a single main so that there is only one entry point, and those who brew install can still use `ratatool [args]` to run commands

Should keep https://github.com/spotify/homebrew-public/pull/24 in sync and merge that PR when this one is merged

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -174,7 +160,6 @@ lazy val ratatoolCli = project
     name := "ratatool-cli",
     libraryDependencies ++= Seq(
       "com.github.scopt" %% "scopt" % scoptVersion,
-      "org.apache.parquet" % "parquet-avro" % parquetVersion,
```
Comment (jbigred1) [inline build.sbt]: did we mean to remove this

File: ratatool-cli/src/main/scala/com/spotify/ratatool/tool/Ratatool.scala [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+object Ratatool {
+
+  //scalastyle:off cyclomatic.complexity
+  def main(args: Array[String]): Unit = {
+    val usage = """
+      | Ratatool - a tool for random data generation and sampling
+      | Usage: ratatool [bigDiffy|bigSampler|directSampler] [args]
+    """.stripMargin
+
+    if (args.isEmpty || !Set("bigDiffy", "bigSampler", "directSampler").contains(args.head)) {
```
Comment (jbigred1) [inline ratatool-cli/src/main/scala/com/spotify/ratatool/tool/Ratatool.scala]: can we make these variables constants

File: ratatool-cli/src/main/scala/com/spotify/ratatool/tool/Ratatool.scala [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import com.spotify.ratatool.io.{AvroIO, BigQueryIO, ParquetIO, TableRowJsonIO}
+import com.spotify.ratatool.samplers.{AvroSampler, BigQuerySampler, BigSampler, ParquetSampler}
+import org.apache.hadoop.fs.Path
+
+object Ratatool {
+
+  //scalastyle:off cyclomatic.complexity
+  def main(args: Array[String]): Unit = {
+    val usage = """
+      | Ratatool - a tool for random data generation and sampling
```
Comment (jbigred1) [inline ratatool-cli/src/main/scala/com/spotify/ratatool/tool/Ratatool.scala]: what about diffing

File: ratatool-cli/src/main/scala/com/spotify/ratatool/tool/Ratatool.scala [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        case "bigDiffy" => BigDiffy.run(args.tail)
+        case "bigSampler" => BigSampler.run(args.tail)
+        case "directSampler" =>
+          val opts = DirectSamplerParser.parse(args.tail)
+          if (opts.isEmpty) {
+            sys.exit(-1)
+          }
+
+          val o = opts.get
+          o.mode match {
```
Comment (jbigred1) [inline ratatool-cli/src/main/scala/com/spotify/ratatool/tool/Ratatool.scala]: instead of `get` could we do `opts.foreach`

**Comment by idreeskhan** [comment]
Yes definitely need to update docs, will roll that into this PR as well

File: build.sbt (hunk 1 above)
Comment (idreeskhan) [reply build.sbt]: Yea I think it ended up there as an artifact during the project restructure but seems to work fine without

File: ratatool-cli/src/main/scala/com/spotify/ratatool/tool/Ratatool.scala (hunk 3 above)
Comment (idreeskhan) [reply ratatool-cli/src/main/scala/com/spotify/ratatool/tool/Ratatool.scala]: Nice catch

File: ratatool-cli/src/main/scala/com/spotify/ratatool/tool/Ratatool.scala (hunk 2 above)
Comment (idreeskhan) [reply ratatool-cli/src/main/scala/com/spotify/ratatool/tool/Ratatool.scala]: 👍

**Pull request merged**